packages/@aws-cdk/aws-appsync/lib/schema-intermediate.ts

```diff
@@ -259,3 +258,88 @@ export class InputType implements IIntermediateType {
     this.definition[options.fieldName] = options.field;
   }
 }
+
+/**
+ * Properties for configuring an Enum Type
+ *
+ * @param definition - the strings that define this enum type
+ *
+ * @experimental
+ */
+export interface EnumTypeProps {
+  /**
+   * the attributes of this type
+   */
+  readonly definition: string[];
+}
+
+/**
+ * Enum Types are abstract types that includes a set of fields
+ * that represent the strings this type can create.
+ *
+ * @experimental
+ */
+export class EnumType implements IIntermediateType {
+  /**
+   * the name of this type
+   */
+  public readonly name: string;
+  /**
+   * the attributes of this type
+   */
+  public readonly definition: { [key: string]: IField };
+
+  public constructor(name: string, props: EnumTypeProps) {
+    this.name = name;
+    this.definition = {};
+    props.definition.map((fieldName: string) => this.addField({ fieldName }));
+  }
+
+  /**
+   * Create an GraphQL Type representing this Enum Type
+   *
+   * @param options the options to configure this attribute
+   * - isList
+   * - isRequired
+   * - isRequiredList
+   */
+  public attribute(options?: BaseTypeOptions): GraphqlType {
+    return GraphqlType.intermediate({
+      isList: options?.isList,
+      isRequired: options?.isRequired,
+      isRequiredList: options?.isRequiredList,
+      intermediateType: this,
+    });
+  }
+
+  /**
+   * Generate the string of this enum type
+   */
+  public toString(): string {
+    return shapeAddition({
+      prefix: 'enum',
+      name: this.name,
+      fields: Object.keys(this.definition),
+    });
+  }
+
+  /**
+   * Add a field to this Enum Type
+   *
+   * Enum Types must have fieldName options.
+   *
+   * @param options the options to add a field
+   */
+  public addField(options: AddFieldOptions): void {
+    if (options.field) {
+      throw new Error('Enum Types does not support IField properties. Use the fieldName option.');
+    }
+    if (!options.fieldName) {
+      throw new Error('The Field option, fieldName, must be declared.');
+    }
+    if (options.fieldName.indexOf(' ') > -1) {
+      throw new Error('Enum fields cannot contain white space.');
```

Comment:
update these error messages to include the expected input and the received. It'll serve as better guidance to users who run into them. 

Think of error messages as help messages. When a user runs into them, we should not only point out what's wrong, but also nudge them in the direction of correcting it.